pkg/virt-controller/watch/vmi.go

```diff
@@ -1097,8 +1097,14 @@ func isPodReady(pod *k8sv1.Pod) bool {
 			return false
 		}
 	}
-
-	return pod.Status.Phase == k8sv1.PodRunning
+	podReadyConditionIsTrue := false
+	for _, condition := range pod.Status.Conditions {
+		if condition.Type == k8sv1.PodReady && condition.Status == k8sv1.ConditionTrue {
+			podReadyConditionIsTrue = true
+			break
+		}
+	}
+	return pod.Status.Phase == k8sv1.PodRunning && podReadyConditionIsTrue
```

Comment:
I think you might be fighting the wrong thing here. Looking at your description, the pod isn't ready due to 
```
    - lastProbeTime: null
      lastTransitionTime: "2024-02-21T08:59:40Z"
      message: corresponding condition of pod readiness gate "kubevirt.io/virtual-machine-unpaused"
        does not exist.
      reason: ReadinessGatesNotReady
      status: "False"
      type: Ready
```

I beleive what's happening is that there's a race condition that prevents the target pod from receiving the "kubevirt.io/virtual-machine-unpaused" condition before the flip from src to target occurs during the live migration.

During migration, we should likely be setting that "kubevirt.io/virtual-machine-unpaused" condition on the pod immediately either before or right as the handoff occurs. Right it's possible there is a tiny gap in time before that condition is set which will result in the target pod temporarily not being ready even after the migration completes.